spec/features/variable_drafts/create_variable_draft_with_associated_collection_spec.rb

```diff
@@ -0,0 +1,159 @@
+
+  before do
+    login(provider: 'SEDAC', providers: %w(SEDAC LARC))
+  end
+
+  context 'when creating a variable draft and associating a collection from the manage variables page' do
+    before do
+      visit manage_variables_path
+    end
+
+    context 'when the supplied concept-id is successfully found' do
+      before do
+        fill_in 'associated_collection_id', with: collection_concept_id
+        click_on 'Create New Record'
+        within '.nav-top' do
+          click_on 'Done'
+        end
+        within '#invalid-draft-modal.eui-modal-content' do
+          click_on 'Yes'
+        end
+      end
+
+      it 'associates the collection upon draft creation' do
+        expect(variable_draft.collection_concept_id).to eq(collection_concept_id)
+      end
+    end
+
+    context 'when the supplied concept-id is not found' do
+      before do
+        fill_in 'associated_collection_id', with: not_found_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'remains on the manage variables page' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+        expect(page).to have_content('Create Variable Record')
+        expect(page).to have_content('(Optional) Enter the Concept ID of the collection to associate this variable:')
+      end
+
+      it 'notifies the user that the concept supplied was not found' do
+        expect(page).to have_content("No matches were found for #{not_found_concept_id}")
+      end
+    end
+
+    context 'when the current provider context is not congruent with the supplied concept-id' do
+      before do
+        fill_in 'associated_collection_id', with: not_current_provider_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'remains on the manage variables page' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+        expect(page).to have_content('Create Variable Record')
+        expect(page).to have_content('(Optional) Enter the Concept ID of the collection to associate this variable:')
+
+      end
+
+      it 'notifies the user that the provider context needs to be changed' do
+        expect(page).to have_content("Variables can only be associated to collections within the same provider. To create a variable for #{not_current_provider_concept_id} you must change your provider context.")
+      end
+    end
+
+    context 'when the cmr response for invalid concept-id is unsuccessful' do
+      before do
+        fill_in 'associated_collection_id', with: invalid_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'surfaces the CMR error message' do
+        expect(page).to have_content('Concept-id [asdf] is not valid.')
+      end
+
+      it 'remains on the manage variables page' do
```

Comment:
You should be checking for some content (like above) that you are on the right page as well.

Reply:
Good catch.